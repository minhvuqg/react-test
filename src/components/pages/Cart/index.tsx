import { FC } from 'react'
import { useSelector } from 'react-redux'

import CartTable from 'components/UI/organisms/CartTable'

import { IProductInCart, StoreType } from 'redux/type.d'
import { IProps } from './type.d'

import './style.scss'

const Cart: FC<IProps> = (): JSX.Element => {
  const productsInCart: IProductInCart[] = useSelector(
    (state: StoreType): IProductInCart[] => state.cart.list
  )

  if (productsInCart.length < 1) {
    return (
      <p className="-text-center -text-medium">
        There are no products in your cart
      </p>
    )
  }

  return (
    <div className="cart-page">
      <p className="-text-medium">Cart</p>
      <div className="-container -card">
        <div>
          <CartTable />
        </div>
      </div>
    </div>
  )
}

export default Cart
