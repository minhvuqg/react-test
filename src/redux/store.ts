import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'

import productSlice from './reducers/productSlice'

const customizedMiddleware: any = getDefaultMiddleware({
  serializableCheck: false
})

const persistedReducer: any = combineReducers({
  cart: productSlice
})

export default configureStore({
  reducer: persistedReducer,
  middleware: customizedMiddleware
})
