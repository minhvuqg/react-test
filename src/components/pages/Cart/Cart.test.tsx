import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import Cart from '../Cart'
import store from 'redux/store'

it('render cart correctly', () => {
  const cartEl = TestRenderer.create(
    < Provider store={store} >
      <Cart />
    </Provider >).toJSON()
  expect(cartEl).toMatchSnapshot()
})
