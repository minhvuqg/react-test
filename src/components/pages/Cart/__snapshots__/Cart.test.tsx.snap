// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render cart correctly 1`] = `
<p
  className="-text-center -text-medium"
>
  There are no products in your cart
</p>
`;
